subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaToolchainBuildOperationsIntegrationTest.groovy

```diff
@@ -409,6 +411,14 @@ class JavaToolchainBuildOperationsIntegrationTest extends AbstractIntegrationSpe
         """
 
         when:
+
+        if (VersionNumber.parse(kotlinPluginVersion) <= VersionNumber.parse("1.6.21")) {
+            executer.expectDeprecationWarning(
+                "The AbstractCompile.destinationDir property has been deprecated. " +
+                "This is scheduled to be removed in Gradle 9.0. " +
+                "Please use the destinationDirectory property instead. " +
+                "Consult the upgrading guide for further information: https://docs.gradle.org/${GradleVersion.current().version}/userguide/upgrading_version_7.html#compile_task_wiring")
```

Comment:
You can use `expectDocumentedDeprecationWarning` and use `/current/` for the version here and the fixture will replace the version in the URL.